Created the SBT contract

gaurangtorvekar opened the pull request:
Responding to OpenOasis issue https://github.com/Cypher-Laboratory/OpenOasis/issues/2

This contract implements an SBT with all the required functionality but uses a dummy signature verification contract because such a contract is not available yet.

gaurangtorvekar committed:
Created the SBT contract

Responding to OpenOasis issue https://github.com/Cypher-Laboratory/OpenOasis/issues/2

This contract implements an SBT with all the required functionality but uses a dummy signature verification contract because such a contract is not available yet.

Elli610 commented on EVM-verifier/contracts/AliceRingToken.sol:
```diff
@@ -0,0 +1,115 @@
+    error OnlyOwnerCanBurn(uint256 tokenId);
+
+    RingSignatureVerifier public verifier; // Instance of the ring signature verifier contract
+    uint256 public minimalAmount; // Minimal amount of tokens to be owned by each address in the ring
```
This should not be a global variable. Each SBT proof can have a different token value. 
`minimalAmount` should be a function variable

Elli610 replied:
it should be an argument of `mint`. Our bad, we forgot it in the contract architecture

Elli610 commented on EVM-verifier/contracts/AliceRingToken.sol:
```diff
@@ -0,0 +1,115 @@
+
+    constructor(address initialOwner, address _verifier, uint256 _minimalAmount)
+        ERC721("AliceRingToken", "ART")
+        Ownable(initialOwner)
```
`Ownable`  does not take any argument

gaurangtorvekar replied:
It does in the latest v5 implementation of Openzeppelin. If you use an earlier version, then there's no argument.
Check this out - https://blog.openzeppelin.com/introducing-openzeppelin-contracts-5.0

Elli610 reviewed (commented):
- I cannot compile the contract with solidity 0.8.20 because `_burn`  should be implemented
- Please add some function documentation on top of each. (using the same template we used in our [contract architecture definition](https://github.com/orgs/Cypher-Laboratory/discussions/2))

Otherwise, LGTM! 
@LeJamon do you see anything else ?

Elli610 reviewed (commented):
closes #2

Elli610 reviewed (changes requested):
changes needed

gaurangtorvekar committed:
fix: Addressed comments from the PR

Changed the @openzeppelin/contracts version to 4.x.x because 5.x.x introduces breaking changes.

gaurangtorvekar commented:
@Elli610 all your comments have been resolved now. Please check...

Elli610 commented on EVM-verifier/contracts/AliceRingToken.sol:
```diff
@@ -0,0 +1,176 @@
+        return false;
+    }
+
+    /**
+     * @notice Verifies the token amounts for each address in the ring
+     *
+     * @param token - the erc20 address of the token we are proving the ownership of
+     * @param signature - the signature object
+     */
+    function verifyTokenAmounts(address token, uint256 minBalance, Signature memory signature)
+        internal
+        returns (bool)
```
- `@param minBalance` is missing
- this function can be restricted to view

gaurangtorvekar replied:
@Elli610 done! Silly that I missed this, thanks for pointing out...

Elli610 reviewed (changes requested):
once `verifyTokenAmounts` is fixed, we should be all good !

Elli610 reviewed (commented):
@LeJamon Is it ok for you ?

gaurangtorvekar committed:
fix: Updated the comment and added view

Addressing comments on the PR
